sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/CollectionResourceManager.java

```diff
@@ -53,45 +57,52 @@ public void createResources() throws CosmosException {
         final CosmosAsyncDatabase database = _client.getDatabase(_configuration.getDatabaseId());
         final CollectionAttributes collectionAttributes = _entityConfiguration.collectionAttributes();
         try {
-            LOGGER.info("Creating container {} in the database {}", containerName, database.getId());
-            final CosmosContainerProperties containerProperties =
-                new CosmosContainerProperties(containerName, Constants.PARTITION_KEY_PATH)
-                    .setIndexingPolicy(collectionAttributes.indexingPolicy());
-            database.createContainerIfNotExists(containerProperties)
-                .block(RESOURCE_CRUD_WAIT_TIME);
+            LOGGER.info("Checking for container {} in the database {}", containerName, database.getId());
+            final CosmosAsyncContainer container = database.getContainer(containerName);
+            final Optional<CosmosContainerResponse> response = getContainerProperties(container);
+            if (response.isPresent()) {
+                LOGGER.info("Container {} already exists in the database {}", containerName, database.getId());
+                applyDesiredContainerPolicies(container, response.get().getProperties());
+                return;
+            }
+
+            createContainer(database, containerName, collectionAttributes);
         } catch (CosmosException e) {
-            LOGGER.error("Exception while creating collection {}", containerName, e);
+            LOGGER.error("Exception while configuring collection {}", containerName, e);
             throw e;
         }
     }
 
     @Override
     public void deleteResources() {
-        deleteExistingCollections();
+        LOGGER.info("The Collection {} will not be deleted.", _configuration.getCollectionId());
+    }
 
-        LOGGER.info("Collection resource cleanup completed");
+    private Optional<CosmosContainerResponse> getContainerProperties(CosmosAsyncContainer container) {
+        try {
+            return Optional.ofNullable(container.read().
+                block(RESOURCE_CRUD_WAIT_TIME));
+        } catch (CosmosException e) {
+            return Optional.empty();
```

Comment:
We should check status code 404 on exception to determine collection does not exist, and for any other exception we should float the exception up

Reply:
I had started down this path, but there are 2 possibilities in the event of an error (and in both cases it returns Optional.empty()
1. Collection does not exist
2. Collection exists, but there was another error

In either case, the code will ahead and call createContainer, which uses createContainerIfNotExists to create the Collection. If there is an error on this call, then we throw the exception all the way up.

To put my long answer differently: What's the impact/consequence of not checking for 404?

Reply:
"Collection exists, but there was another error" in this case error could be anything so it is ideal to float the error right there instead of going further in the logic. In this case it is not a big deal as this is testing workload, so your call. But it is advisable to do the right error check in real prod application.

Reply:
I agree if there is an error, there's not much value in continuing.